web/impact/impact/v1/helpers/profile_helper.py

```diff
@@ -506,10 +508,15 @@ def is_non_member(self):
     def confirmed_user_program_families(self):
         prg = _confirmed_non_future_program_role_grant(self.subject)
         program_ids = _latest_program_id_foreach_program_family()
-        program_families = list(prg.filter(
-            program_role__program__pk__in=program_ids
-        ).values_list(
-            'program_role__program__program_family__name', 'created_at'))
+        program_families = list(
```

Comment:
The reason we are including `person__date_joined` is non-obvious to the average observer. I think a short comment at the top of this method would be a kindness. E.g. 
`# We include person__date_joined as a fallback for legacy PRGs with no date`

Also, this is bikeshedding but `program_family_data` would be a less misleading name for the working variable, since it now contains non-program-family data.  (Likewise `confirmed_user_program_data` would be an improved name for this method.)